tests/editing/thousand_pounds/tests/test_thousand_pounds.py

```diff
@@ -1,338 +1,805 @@
-            "Attribute 'lower_limit' is missing or not a valid number",
-        ),
-        id="Invalid format for lower_limit variable",
-    ),
-    pytest.param(
-        "Dummy",
-        10,  # principal
-        56,  # predictive
-        23,  # auxiliary
-        350,  # upper limit
-        351,  # lower limit
-        [],
-        Thousands_output(
-            "Dummy",
-            10,
-            10,
-            [],
-            None,
-            "E",
-            "Lower limit is larger than the upper limit (351:350)",
-        ),
-        id="Lower limit is larger than the upper limit",
-    ),
-]
+    )
+    def test__run__given_valid_config__returns_adjusted_figures(
+        self,
+        principal_identifier,
+        principal_variable,
+        predictive,
+        auxiliary,
+        upper_limit,
+        lower_limit,
+        target_variables,
+        expected,
+        test_id,
+    ):
+        if isinstance(expected, Thousands_output):
+            try:
+                result = run(
+                    principal_identifier,
+                    principal_variable,
+                    predictive,
+                    auxiliary,
+                    upper_limit,
+                    lower_limit,
+                    target_variables,
+                )
+                assert result == expected
 
+            except Exception as e:
+                pytest.fail(
+                    EXCEPTION_FAIL_MESSAGE.format(
+                        test_id=test_id,
+                        exception_type=type(e).__name__,
+                        exception_msg=str(e),
+                    )
+                )
+        else:
+            with pytest.raises(Exception) as exc_info:
+                run(
+                    principal_identifier,
+                    principal_variable,
+                    predictive,
+                    auxiliary,
+                    upper_limit,
+                    lower_limit,
+                    target_variables,
+                )
+            assert (str(exc_info.value)) == str(expected)
 
-@pytest.mark.parametrize(
-    "principal_identifier, principal_variable, predictive, auxiliary, upper_limit, lower_limit, target_variables, expected",
-    testdata,
-)
-def test__run__given_valid_config__returns_adjusted_figures(
-    principal_identifier,
-    principal_variable,
-    predictive,
-    auxiliary,
-    upper_limit,
-    lower_limit,
-    target_variables,
-    expected,
-):
-    assert (
-        run(
-            principal_identifier,
-            principal_variable,
-            predictive,
-            auxiliary,
-            upper_limit,
-            lower_limit,
-            target_variables,
-        )
-        == expected
+
+class TestValidateInput:
+    @pytest.mark.parametrize(
+        "predictive, auxiliary, principal_variable, lower_limit, upper_limit, target_variables, expected_result, "
+        "test_id",
+        [
+            (
+                "500",
```

Comment:
Could you add a comment to explain what each element relates to... this example is the main test from above:

"q100",
                50000000,  # principal value/variable
                50000,  # predictive
                15000,  # auxiliary
                1350,  # upper limit
                350,  # lower limit